edx/edx-enterprise
# ENT-317 Add externally managed consent option for data sharing

**Description:** For integrations like SAP, consent to share data back has already been gathered by the partner, and to simplify the experience for users, we want to skip prompting for consent while still recording and collecting data to send back to the partner.

**JIRA:** https://openedx.atlassian.net/browse/ENT-317

**Dependencies:** n/a

**Merge deadline:** 5/5/2017

**Installation instructions:** 

**Testing instructions:**

Sandbox: https://brittneyexline.sandbox.edx.org/dashboard

1. Login as staff to django admin, configure an enterprise customer (Bexline Inc recommended on the above sandbox) to have data sharing consent enabled and enforces data sharing consent to be externally managed. Save and logout.
2. Go to the registration page and sign in using the TestShib IDP, which should be linked to the enterprise customer. Finish the registration flow through to the dashboard.
3. SSH to the machine and using the mysql command line tool, verify that a new record has been added to enterprise_userdatasharingconsentaudit with state = 'externally_managed'.
4. Bonus (but not properly configured on my sandbox because discovery is down): Go to a tpa-hinted track selection page with a verified option for the course. Go through the verified course option all the way to checkout and the receipt page. You should not be prompted for consent at any point, and a new Enterprise Enrollment record should be written.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** I want to note that this option will only work, and only makes sense, if SSO is configured for the enterprise customer. This PR was made on that assumption.
